nebula_decoders/src/nebula_decoders_velodyne/decoders/vlp16_decoder.cpp

```diff
@@ -94,12 +94,43 @@ void Vlp16Decoder::reset_pointcloud(size_t n_pts, [[maybe_unused]] double time_s
   scan_timestamp_ = -1;
 }
 
-void Vlp16Decoder::reset_overflow([[maybe_unused]] double time_stamp)
+void Vlp16Decoder::reset_overflow(double time_stamp)
 {
+  if (overflow_pc_->points.size() == 0) {
+    scan_timestamp_ = -1;
+    overflow_pc_->points.reserve(max_pts_);
+    return;
+  }
+
+  // Compute the absolute time stamp of the last point of the overflow pointcloud
+  const double last_overflow_time_stamp =
+    scan_timestamp_ + 1e-9 * overflow_pc_->points.back().time_stamp;
+
+  // Detect cases where there is an unacceptable time difference between the last overflow point and
+  // the first point of the next packet. In that case, there was probably a packet drop so it is
+  // better to ignore the overflow pointcloud
+  if (time_stamp - last_overflow_time_stamp > 0.05) {
+    scan_timestamp_ = -1;
+    overflow_pc_->points.clear();
+    overflow_pc_->points.reserve(max_pts_);
```

Comment:
Here there should be a return. I actually forgot it in my PR (I swear it was there !)

Reply:
Fixed in 638c532f6838975ff87846d1f11751a6233a252b